subsys/net/lib/download_client/src/http.c

```diff
@@ -182,19 +184,19 @@ static int http_header_parse(struct download_client *client, size_t *hdr_len)
 	 */
 	if (client->file_size == 0) {
 		if (using_range_requests) {
-			p = strstr(client->buf, "content-range");
+			p = strnstr(client->buf, "content-range", sizeof(client->buf));
 			if (!p) {
 				LOG_ERR("Server did not send "
 					"\"Content-Range\" in response");
 				return -1;
 			}
-			p = strstr(p, "/");
+			p = strnstr(p, "/", sizeof(client->buf));
```

Comment:
`p` is already at some offset, so this could still reach outside the buffer.

Reply:
Thanks

Reply:
Should be okay now